service/src/test/java/skills/intTests/saml2/SAML2IntegrationIT.groovy

```diff
@@ -0,0 +1,70 @@
+/**
+ * Copyright 2020 SkillTree
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package skills.intTests.saml2
+
+import dasniko.testcontainers.keycloak.KeycloakContainer
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.http.HttpStatus
+import org.springframework.boot.test.context.SpringBootTest
+import org.springframework.boot.test.web.server.LocalServerPort
+import org.springframework.boot.test.web.client.TestRestTemplate
+import org.springframework.http.ResponseEntity
+import skills.SpringBootApp
+import spock.lang.Shared
+import spock.lang.Specification
+
+
+@SpringBootTest(webEnvironment=SpringBootTest.WebEnvironment.RANDOM_PORT, classes = SpringBootApp)
+class SAML2IntegrationIT extends Specification{
+
+    @LocalServerPort
+    int port
+
+    @Autowired
+    private TestRestTemplate restTemplate
+
+    @Shared
+    KeycloakContainer keycloak = new KeycloakContainer().withRealmImportFile("keycloak/realm-export.json");
+
+
+    def setupSpec() {
+        keycloak.start()
+
+        // Set the SAML IDP metadata for the Keycloak instance
```

Comment:
Please move configuration inside` @SpringBootTest(properties = [])` on line 30; see `skills.intTests.RankingAndProgressViewsDisabledIT` for an example

Reply:
Some of the properties like application port are dynamic so could not place those in properties array